### PR TITLE
Add multisource configuration

### PR DESCRIPTION
**Pull request by pdambrauskas**
@Minutis 

Configuration source priorities are defined by configurationSource list order.
Possible sources:
- args
- yaml
- env

~~nested structures are supported by yaml only~~
~~missing property validation (should be easy to implement with javax annotations)~~

### CODE REVIEW

**Comment by Minutis** [comment]
This means that you can provide properties to racoon when submitting a command through different sources (yaml, env, cli)?

File: racoon-node/src/main/java/run/racoon/node/RacoonNode.java [hunk 1]
```diff
@@ -1,36 +1,54 @@
...
 
+import java.io.File;
+import java.util.ArrayList;
 import java.util.Collections;
 import java.util.List;
 
 public class RacoonNode {
     private static final Logger LOG = LoggerFactory.getLogger(RacoonNode.class);
 
     public static void main(String[] args) {
```
Comment (Minutis) [inline racoon-node/src/main/java/run/racoon/node/RacoonNode.java]: Should i review main method? Seems a bit messy, but maybe it's too early to review?

Comment (pdambrauskas) [reply racoon-node/src/main/java/run/racoon/node/RacoonNode.java]: Lets leave main method for later, as it has no tests and does nothing : )

**Comment by pdambrauskas** [comment]
> This means that you can provide properties to racoon when submitting any command through different sources (yaml, env, cli)?

Yes! You can provide path to config file, configs themselves, or do everything on ENV variables :)

**Pull request merged**